src/puppetlabs/puppetdb/http/event_counts.clj

```diff
@@ -27,8 +27,10 @@
 
 (defn event-counts-app
   "Ring app for querying for summary information about resource events."
-  [version]
-  (log/warn "The event-counts endpoint is experimental and may be altered or removed in the future.")
+  [version globals]
+  (when-not (= "puppetdb" (:product-name globals))
```

Comment:
would moving this check into routes section like in aggregate-event-counts allow you to avoid parameterizing v4-app?